### PR TITLE
[labs/analyzer] Add separate entrypoint for createPackageAnalyzer

### PR DESCRIPTION
**Pull request by justinfagnani**
Previously `index.ts` imported both the main `Analyzer` class and `createPackageAnalyzer()`, which uses Node APIs.

This PR separates the entrypoints into `index.ts` which only exports the Analyzer class and model interfaces, and `package-analyzer.ts` which exports all of that, plus `createPackageAnalyzer()`.

This is a breaking change.

### CODE REVIEW

File: packages/labs/analyzer/package.json [hunk 1]
```diff
@@ -53,7 +53,8 @@
     "!/lib/.tsbuildinfo"
   ],
   "exports": {
-    ".": "./index.js"
+    ".": "./index.js",
+    "./package-analyzer.js": "./package-analyzer.js"
```
Comment (rictic) [inline packages/labs/analyzer/package.json:57]: package-analyzer.js or analyze-package.js?

**Review by rictic — CHANGES_REQUESTED** [review]
Build is failing

**Review by kevinpschaaf — APPROVED** [review]
LGTM, modulo the naming issue @rictic raised. Did you want the top-level import to be `package-analyzer.js` to match `analyzer.js`? If so, need to rename the file, otherwise rename the `exports` in package.json.

**Comment by justinfagnani** [comment]
Argh. I don't understand why the build is failing on CI now. Build and tests are passing on my machine even after a git clean. I also don't understand why it was _passing_ on my machine earlier as I had some obviously bad imports.

**Review by rictic — APPROVED** [review]

**Pull request merged**